app/scripts/models/user.js

```diff
@@ -733,6 +733,46 @@ define(function (require, exports, module) {
      */
     rejectAccountUnblockCode(account, unblockCode) {
       return account.rejectUnblockCode(unblockCode);
+    },
+
+    /**
+     * Should the model be initialized using browser data?
+     *
+     * @param {Object} service service being signed into.
+     * @returns {Boolean}
+     */
+    shouldSetSignedInAccountFromBrowser (service) {
+      // If service=sync, always use the browser's state of the world.
+      // If trying to sign in to an OAuth relier, prefer any users that are
+      // stored in localStorage and only use the browser's state if no
+      // user is stored.
+      return !! (service === Constants.SYNC_SERVICE || this.getSignedInAccount().isDefault());
```

Comment:
Looks like the explicit `!!` is unnecessary here? `===` returns a boolean and so does `isDefault()`. Not a big deal obviously but may be a trifle more readable without it?